tests/test_pipdeptree.py

```diff
@@ -488,3 +490,25 @@ def test_custom_interpreter(tmp_path, monkeypatch, capfd, args_joined):
     assert context.value.code == 1
     assert not out
     assert err == "graphviz functionality is not supported when querying" " non-host python\n"
+
+
+def test_guess_version_setuptools():
+    code = """
+    import sys
+
+    import pipdeptree
+
+    if sys.version_info >= (3, 8):
+        import importlib.metadata as importlib_metadata
+    else:
+        import importlib_metadata
+
+    def raise_import_error(name):
+        raise ImportError(name)
+
+    importlib_metadata.version = raise_import_error
+
+    print(pipdeptree.guess_version("setuptools"), end="")
+    """
```

Comment:
You loose code coverage and debug-ability this way. Make it a side file instead, and call the python file via sys.executable instead of -c